api/custom_metadata/views.py

```diff
@@ -0,0 +1,64 @@
+from django.http import Http404
+import rest_framework
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from api.base import permissions as base_permissions
+from api.base.views import JSONAPIBaseView
+from api.nodes.permissions import ContributorOrPublic
+
+import osf.models as osfdb
+from api.base.permissions import PermissionWithGetter
+from .serializers import CustomFileMetadataSerializer, CustomItemMetadataSerializer
+
+
+class CustomFileMetadataDetail(JSONAPIBaseView, rest_framework.generics.RetrieveUpdateAPIView):
+    permission_classes = (
+        PermissionWithGetter(ContributorOrPublic, 'guid.referent.target'),
+        rest_framework.permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.GUIDS_READ]  # TODO
+    required_write_scopes = [CoreScopes.NULL]  # TODO
+
+    serializer_class = CustomFileMetadataSerializer
+    view_category = 'custom-metadata'
+    view_name = 'custom-file-metadata-detail'
+
+    def get_object(self):
+        try:
+            metadata_record = osfdb.GuidMetadataRecord.objects.for_guid(
+                self.kwargs['guid_id'],
+                allowed_referent_models=(osfdb.BaseFileNode,),
+            )
+        except osfdb.base.InvalidGuid:
+            raise Http404
+        self.check_object_permissions(self.request, metadata_record)
+        return metadata_record
+
+
+class CustomItemMetadataDetail(JSONAPIBaseView, rest_framework.generics.RetrieveUpdateAPIView):
+    permission_classes = (
+        PermissionWithGetter(ContributorOrPublic, 'guid.referent'),
```

Comment:
Jon and I are tending to towards just write an entirely new permission in these cases, it makes it easier it individualize them per view if they need to be changed.

Reply:
huh, i'd have thought to go the other direction, try to make permissions more consistent with fewer permission classes reused more often -- but sure can add bespoke permission here